internal/execute/tsc.go

```diff
@@ -241,7 +264,28 @@ func performCompilation(
 		JSDocParsingMode: ast.JSDocParsingModeParseForTypeErrors,
 	})
 	parseTime := sys.Now().Sub(parseStart)
+	return emitAndReportStatistics(
+		sys,
+		program,
+		func() *compiler.Program {
+			return program
+		},
+		config,
+		reportDiagnostic,
+		configTime,
+		parseTime,
+	)
+}
 
+func emitAndReportStatistics(
+	sys System,
+	program compiler.AnyProgram,
+	getCoreProgram func() *compiler.Program,
```

Comment:
Is this indirection needed? Can't just be immediately invoked? Or just use `program`?